### PR TITLE
Update `filter by price` skeleton design

### PR DESCRIPTION
**Pull request by albarin**
This PR updates the `Filter by price` loading state according to the new designs.

Related to https://github.com/woocommerce/woocommerce-blocks/issues/6846

| Block | Loading |
| ------ | ----- |
| Screenshot 2022-08-29 at 15 00 51| Screenshot 2022-08-29 at 15 00 55|
| Screenshot 2022-08-29 at 15 01 15| Screenshot 2022-08-29 at 15 01 28|

#### User Facing Testing

1. Enable TT2 theme. Create a new page with an `All Products` block and a `Filter by price` block. Save the page.
2. Open the page in the frontend and check that the `Filter by price` block shows like the screenshot above while loading.
3. Try different combinations of settings (show the `Apply` button, make the inputs inline, and make the range price not editable, etc) and make sure the loading state makes sense and there are no regressions.
4. Repeat steps 1-3 with the Storefront theme.

### WooCommerce Visibility

* [X] WooCommerce Core
* [X] Feature plugin
* [ ] Experimental

> Update the Filter by Price loading state design.

### CODE REVIEW

**Comment by albarin** [comment]
@vivialice could you try it as well and give some feedback 🙏

File: assets/js/base/components/price-slider/index.tsx [hunk 1]
```diff
@@ -33,6 +33,7 @@ export interface PriceSliderProps {
 	 * Whether values are loading or not.
 	 */
 	isLoading?: boolean;
+	isUpdating?: boolean;
```
Comment (tjcafferkey) [inline assets/js/base/components/price-slider/index.tsx:39]: Can we put a comment here with some description on what differentiates `isLoading` from `isUpdating`?

**Comment by vivialice** [comment]
Hmm it is showing up like this for me? I can see the title when loading. 

Screenshot 2022-08-30 at 14 27 56

with 'apply' button:
Screenshot 2022-08-30 at 14 29 34

**Comment by dinhtungdu** [comment]
@albarin The animation of the placeholder is a bit weird to me:

https://user-images.githubusercontent.com/5423135/187915175-6f8d4935-efad-4032-a040-c062206e41a6.mov

I notice the background color at https://github.com/woocommerce/woocommerce-blocks/blob/update/6846-filter-price-skeleton/assets/js/base/components/filter-placeholder/style.scss#L3 causes the issue. Removing/commenting that line fixes the animation: 

https://user-images.githubusercontent.com/5423135/187915477-b5244c09-75a6-442b-a6e4-c9a94ee3ca78.mov

**Comment by dinhtungdu** [comment]
@albarin Also, look like it enabling the Apply filter button doesn't have any effect on the placeholder, is that expected?


https://user-images.githubusercontent.com/5423135/187920108-2214acf0-8e10-4703-bbcf-6d4c94d35307.mov

**Comment by albarin** [comment]
> @albarin The animation of the placeholder is a bit weird to me

Thanks @dinhtungdu ! I pushed a fix, let me know if it looks better now.

**Comment by albarin** [comment]
> @albarin Also, look like it enabling the Apply filter button doesn't have any effect on the placeholder, is that expected?
> Screen.Recording.2022-09-01.at.20.00.26.mov

@vivialice could you help us here? Should the Apply button also have a placeholder? 🤔

**Review by dinhtungdu — COMMENTED** [review]
@albarin the fix looks good to me! 🎉 

But I have a concern related to the layout shift issue:

Currently, we only load the skeleton on client-side, after the block is rendered on the front end. It means that:

- The block is empty on page load.
- The layout shift will occur when the block is rendered and the skeleton appears.

Can we do something to prevent that, like render the skeleton server-side or save the skeleton to the database?

Also, I think that we're changing too many components to include the skeleton, which makes those components more complex and hard to maintain in the future. I'm wondering if can we relook at [this approach](https://github.com/woocommerce/woocommerce-blocks/pull/6877#issuecomment-1211543847)? We can now solve the block recovery issue by using block deprecation.

**Review by dinhtungdu — APPROVED** [review]
LGTM! As discussed in the chat, we can merge this and then address the server-side rendering later. The skeleton for the filter button can also be implemented if needed in a separate PR for the price, stock, and attribute filter blocks at once.

**Pull request merged**